tests/unittests/Foundation/NSNotificationTests.mm

```diff
@@ -0,0 +1,32 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <TestFramework.h>
+#import <Foundation/Foundation.h>
+@interface TestObjectA : NSObject
+@end
+
+@implementation TestObjectA
+- (void)dealloc {
+    [[NSNotificationCenter defaultCenter] postNotificationName:@"MOOSENOTIFICATION" object:self];
+    [super dealloc];
+}
+
+@end
+
+TEST(NSNotificationCenter, PostNotificationFromDealloc) {
+    TestObjectA* obj = [[TestObjectA new] autorelease];

```

Comment:
I have tested it with a release, but this is the test code from the issue report, so i decided to leave the autorelease version in.

---
In reply to: [69334437](https://github.com/Microsoft/WinObjC/pull/600#discussion_r69334437) [](ancestors = 69334437)